src/main/io/serial_4way.c

```diff
@@ -464,36 +463,36 @@ void esc4wayProcess(serialPort_t *mspPort)
         // restart looking for new sequence from host
         do {
             CRC_in.word = 0;
-            timedOut = ReadByteCrc(&ESC, ESC_TIMEOUT_US);
-        } while ((ESC != cmd_Local_Escape) && !timedOut);
+            // No timeout as BLHeliSuite32 has this loops sitting indefinitely waiting for input
+            ReadByteCrc(&ESC, 0);
+        } while (ESC != cmd_Local_Escape);
 
         RX_LED_ON;
 
         Dummy.word = 0;
         O_PARAM = &Dummy.bytes[0];
         O_PARAM_LEN = 1;
 
+        timedOut = ReadByteCrc(&CMD, CMD_TIMEOUT_US);
+        timedOut |= ReadByteCrc(&ioMem.D_FLASH_ADDR_H, ARG_TIMEOUT_US);
+        timedOut |= ReadByteCrc(&ioMem.D_FLASH_ADDR_L, ARG_TIMEOUT_US);
+        timedOut |= ReadByteCrc(&I_PARAM_LEN, ARG_TIMEOUT_US);
+
         if (!timedOut) {
```

Comment:
```
  for (int i = 0; !timeout && i < I_PARAM_LEN; i++)
     timedOut |= ReadByteCrc(ParamBuff + i, DAT_TIMEOUT_US);
  for (int i = 1; !timeout && i >= 0; i++)
      timedOut |= ReadByte(&CRC_check.bytes[i], CRC_TIMEOUT_US);

Reply:
Thanks. I've implemented a change inspired by this.

Reply:
> I prefer at the end. Where in our coding guidelines does it suggest otherwise?

the `||` ?  if yes, i'm sorry i committed the suggestion. please revert it.  also this comment in the wrong section. (> https://github.com/betaflight/betaflight/pull/13287#discussion_r1450540298)